dbms/src/Functions/FunctionsIsIPAddr.h

```diff
@@ -0,0 +1,288 @@
+    unsigned char tmp[16], *tp, *endp, *colonp;
+    const char *xdigits, *curtok;
+    int ch, saw_xdigit;
+    unsigned int val;
+
+    memset((tp = tmp), '\0', IN6ADDRSZ);
+    endp = tp + IN6ADDRSZ;
+    colonp = nullptr;
+    if (*src == ':')
+        if (*++src != ':')
+            return 0;
+    curtok = src;
+    saw_xdigit = 0;
+    val = 0;
+    while ((ch = *src++) != '\0')
+    {
+        const char * pch;
+
+        if ((pch = strchr((xdigits = xdigits_l), ch)) == nullptr)
+            pch = strchr((xdigits = xdigits_u), ch);
+        if (pch != nullptr)
+        {
+            val <<= 4;
+            val |= (pch - xdigits);
+            if (val > 0xffff)
+                return 0;
+            saw_xdigit = 1;
+            continue;
+        }
+        if (ch == ':')
+        {
+            curtok = src;
+            if (!saw_xdigit)
+            {
+                if (colonp)
+                    return 0;
+                colonp = tp;
+                continue;
+            }
+            if (tp + INT16SZ > endp)
+                return 0;
+            *tp++ = (unsigned char)(val >> 8) & 0xff;
+            *tp++ = (unsigned char)val & 0xff;
+            saw_xdigit = 0;
+            val = 0;
+            continue;
+        }
+        if (ch == '.' && ((tp + INADDRSZ) <= endp) && isIPv4(curtok) > 0)
+        {
+            tp += INADDRSZ;
+            saw_xdigit = 0;
+            break; /* '\0' was seen by isIPv4(). */
+        }
+        return 0;
+    }
+    if (saw_xdigit)
+    {
+        if (tp + INT16SZ > endp)
+            return 0;
+        *tp++ = (unsigned char)(val >> 8) & 0xff;
+        *tp++ = (unsigned char)val & 0xff;
+    }
+    if (colonp != nullptr)
+    {
+        const size_t n = tp - colonp;
+        size_t i;
+
+        for (i = 1; i <= n; i++)
+        {
+            endp[-i] = colonp[n - i];
+            colonp[n - i] = 0;
+        }
+        tp = endp;
+    }
+    if (tp != endp)
+        return 0;
+    return 1;
+}
+
+class FunctionIsIPv4 : public IFunction
+{
+public:
+    static constexpr auto name = "tiDBIsIPv4";
+    FunctionIsIPv4() = default;
+
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionIsIPv4>(); };
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 1)
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be 1.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        return std::make_shared<DataTypeUInt8>();
+    }
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        if (const auto * col_input = checkAndGetColumn<ColumnString>(block.getByPosition(arguments[0]).column.get()))
```

Comment:
const column should be checked, or you can override `useDefaultImplementationForConstants` and return true

Reply:
done